proposals/044-integration-model.md

```diff
@@ -0,0 +1,203 @@
+
+## State reconciliation 
+
+Integrations follow a declarative reconciliation paradigm. An integration is always in a _currentState_. As consequence of an _action_ (see below) a new _targeState_ can be set. It is now the duty of the Syndesis backend to reconcile the _currentState_ to become the _targetState_.
+This modelled is closely to the Kubernetes state model itself.
+
+Reconciliation itself can be in one of two states:
+
+* _Pending_ when `currentState != targetState`
+* _Ok_ when `currentState == targetState`
+
+This state is modelled implicitely by looking at both state variables. 
+
+`Error` can never be a `targetState` but it can be a `currentState`. The backend controllers try to get out of the error state but might give up at some point. 
+
+Every state change (current or target) should be tracked. 
+
+## Operations
+
+The following operations change the `targetState`. The `targetState` could be updated by simply changing the Integration's field, but it is recommended to use a more semantic rich REST API call for perfoming this actions (i.e. having specific endpoint which can be used by the UI to trigger the targetState change).
+
+* **Publish** sets the target state to _Active_. This operation can be called from the states:
+  - _Draft_ when the integration should be published for the first time
+  - _Inactive_ when the integration has been stopped
+  - _Error_ when an integration should be re-deployed in case of an error to trigger a retry
+
+* **Suspend** sets the target state to _Inactive_. This operation should be only be possible if the `currentState` is _Active_.
+
+* **Resume** is an operation which can be used when the `currentState` is _Inactive_ and the integration should be re-activated. This sets the `targetState` to _Active_
+
+* **Undeploy** marks an integration target state as _Undeployed_. An integration should be _Inactive_ before the _Undelpoyed_ state can be reached.
+
+* **Delete** is the operation which removes an integration in an _Draft_ state.  
+
+The `syndesis-rest` backend in this case will simply validate the context (i.e. whether the operation is allowed), updates the `targetState` and the returns to `syndesis-ui`. 
+A background controller detects the state change and performs the proper actions to reach the target state. 
+
+## Versioning
+
+Each integration has a unique name within a certain context (which currently is the whole installation). 
+Multiple versions of an integration can exist. 
+A _active_ integration is immutable.
+So in order to update an active integration (i.e. an integration which is in state _Active_) a new version needs to be created.
+Changing an _active_ integration in the UI causes a new integration to be created, with the following characteristics:
+* Its cloned from the _active_ integration
+* The state of the clone is set to _Draft_
+* The `parentVersion` is set to the version of the active integration it is cloned from
+* The `version` is set to undefined.
```

Comment:
Agreed, I'm going to rearrange the document a bit to clarify the difference between "Save" and "Publish".